bson/extjson_reader.go

```diff
@@ -27,19 +27,19 @@ type extJSONValueReader struct {
 }
 
 // NewExtJSONValueReader creates a new ValueReader from a given io.Reader
-// It will interpret the JSON of r as canonical or relaxed according to the
-// given canonical flag
-func NewExtJSONValueReader(r io.Reader, canonical bool) (ValueReader, error) {
-	return newExtJSONValueReader(r, canonical)
+// It will interpret the JSON of r as canonical only, or allow both
+// canonical and relaxed according to the given canonicalOnly flag
```

Comment:
I recommend rewording this comment to match the recommended comment style on `UnmarshalExtJSON`:
```go
// NewExtJSONValueReader returns a ValueReader that reads Extended JSON values
// from r. If canonicalOnly is true, reading values from the ValueReader returns
// an error if the Extended JSON was not marshaled in canonical mode.
```